addons/sale/models/res_partner.py

```diff
@@ -43,3 +43,20 @@ def can_edit_vat(self):
             ('state', 'in', ['sent', 'sale', 'done'])
         ], limit=1)
         return can_edit_vat and not bool(has_so)
+
+    @api.depends_context('company', 'sale_order_ids')
+    def _credit_debit_get(self):
+        # Extended in order to include the amount of confirmed sales orders in the partner's credit,
+        # after withdrawing what has already been invoiced.
+        super()._credit_debit_get()
+        residual_amounts = self.env['sale.order'].read_group(
+            domain=[('state', '=', 'sale'), ('company_id', '=', self.env.company.id)],
+            fields=['partner_id', 'amount_residual:array_agg', 'currency_rate:array_agg'],
+            groupby=['partner_id']
+        )
+        residual_amount_by_partner = {
+            row['partner_id'][0]: sum([x * y for x, y in zip(row['amount_residual'], row['currency_rate'])])
+            for row in residual_amounts
+        }
```

Comment:
@Feyensv @william-andre 
I am wondering if it would not be better to have an `amount_residual_currency` field stored on `sale.order` to avoid this computation ?
Or  could replace `amount_residual` by `amount_residual_currency` altogether, but I thought `amount_residual` might be useful outside this feature ?
I don't know what is better : having to store an additional column or having to compute the sum of the products every time :thinking: 
Would really appreciate your insight on this :slightly_smiling_face:

Reply:
Fetching all the sales orders is values is probably not the best idea indeed. A SQL query doing that wouldn't be too hard to write.
As for storing the residual amount in company currency, I'm not sure it is useful if the rate is already stored. It would only be a small step in the aggregation of the SUM in this case.